drivers/i2c/i2c_omap.c

```diff
@@ -0,0 +1,815 @@
+	uint8_t RESERVED_3[0x4C];    /**< Reserved, offset: 0x38 - 0x84 */
+	__IO uint32_t IE;            /**< Interrupt Enable (Legacy), offset: 0x84 */
+	__IO uint32_t STAT;          /**< Status, offset: 0x88 */
+	uint8_t RESERVED_4[0x4];     /**< Reserved, offset: 0x8C - 0x90 */
+	__IO uint32_t SYSS;          /**< System Status, offset: 0x90 */
+	__IO uint32_t BUF;           /**< Buffer, offset: 0x94 */
+	__IO uint32_t CNT;           /**< Data Count, offset: 0x98 */
+	__IO uint32_t DATA;          /**< Data Access, offset: 0x9C */
+	uint8_t RESERVED_5[0x4];     /**< Reserved, offset: 0xA0 - 0xA4 */
+	__IO uint32_t CON;           /**< Configuration, offset: 0xA4 */
+	__IO uint32_t OA;            /**< Own Address, offset: 0xA8 */
+	__IO uint32_t SA;            /**< Target Address, offset: 0xAC */
+	__IO uint32_t PSC;           /**< Clock Prescaler, offset: 0xB0 */
+	__IO uint32_t SCLL;          /**< SCL Low Time, offset: 0xB4 */
+	__IO uint32_t SCLH;          /**< SCL High Time, offset: 0xB8 */
+	__IO uint32_t SYSTEST;       /**< System Test, offset: 0xBC */
+	__IO uint32_t BUFSTAT;       /**< Buffer Status, offset: 0xC0 */
+} i2c_omap_regs_t;
+
+/* I2C Interrupt Enable Register (I2C_OMAP_IE) */
+#define I2C_OMAP_IE_XDR  BIT(14) /* TX Buffer drain interrupt enable */
+#define I2C_OMAP_IE_RDR  BIT(13) /* RX Buffer drain interrupt enable */
+#define I2C_OMAP_IE_XRDY BIT(4)  /* TX data ready interrupt enable */
+#define I2C_OMAP_IE_RRDY BIT(3)  /* RX data ready interrupt enable */
+#define I2C_OMAP_IE_ARDY BIT(2)  /* Access ready interrupt enable */
+#define I2C_OMAP_IE_NACK BIT(1)  /* No acknowledgment interrupt enable */
+#define I2C_OMAP_IE_AL   BIT(0)  /* Arbitration lost interrupt enable */
+
+/* I2C Configuration Register (I2C_OMAP_CON) */
+#define I2C_OMAP_CON_EN        BIT(15) /* I2C module enable */
+#define I2C_OMAP_CON_OPMODE_HS BIT(12) /* High Speed support */
+#define I2C_OMAP_CON_MST       BIT(10) /* Controller/target mode */
+#define I2C_OMAP_CON_TRX       BIT(9)  /* TX/RX mode (controller only) */
+#define I2C_OMAP_CON_STP       BIT(1)  /* Stop condition (controller only) */
+#define I2C_OMAP_CON_STT       BIT(0)  /* Start condition (controller) */
+
+/* I2C_OMAP_WE wakeup enable register */
+#define I2C_OMAP_WE_XDR_WE  BIT(14) /* TX drain wakeup */
+#define I2C_OMAP_WE_RDR_WE  BIT(13) /* RX drain wakeup */
+#define I2C_OMAP_WE_AAS_WE  BIT(9)  /* Address as target wakeup */
+#define I2C_OMAP_WE_BF_WE   BIT(8)  /* Bus free wakeup */
+#define I2C_OMAP_WE_STC_WE  BIT(6)  /* Start condition wakeup */
+#define I2C_OMAP_WE_GC_WE   BIT(5)  /* General call wakeup */
+#define I2C_OMAP_WE_DRDY_WE BIT(3)  /* TX/RX data ready wakeup */
+#define I2C_OMAP_WE_ARDY_WE BIT(2)  /* Register access ready wakeup */
+#define I2C_OMAP_WE_NACK_WE BIT(1)  /* No acknowledgment wakeup */
+#define I2C_OMAP_WE_AL_WE   BIT(0)  /* Arbitration lost wakeup */
+
+#define I2C_OMAP_WE_ALL                                                                            \
```

Comment:
This define is only used once, just use the content in the one place it is used.

Speaking on that one place, do we know if it is needed? The wakeup support is probably not functional right now, you could just drop this an all the `I2C_OMAP_WE*` definitions in that case.

Reply:
I have removed it!